flink-test-utils-parent/flink-test-utils/pom.xml

```diff
@@ -117,6 +124,41 @@ under the License.
 				<inherited>true</inherited>
 				<extensions>true</extensions>
 			</plugin>
+
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>shade-flink</id>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<artifactSet>
+								<includes combine.children="append">
+									<include>io.netty:netty</include>
+								</includes>
+							</artifactSet>
+							<relocations combine.children="append">
+								<relocation>
+									<pattern>org.jboss.netty</pattern>
+									<shadedPattern>org.apache.flink.shaded.testutils.org.jboss.netty</shadedPattern>
```

Comment:
Could you explain more why we need to shade this here? are we hiding some transitive netty dependency?

Reply:
Its not a transitive dependency, but a direct one. Test-utils depends on netty directly for the network connection proxy. Because Netty clashes often and such clashes have made tests unstable in a very subtle way, I wanted to hide netty here again to avoid such future problems

Reply:
ah ok. How come we aren't using flink-shaded-netty for that?

Reply:
Good question, maybe @pnowojski can answer that... My guess is he was more familiar with the Netty 3 API (org.jboss.netty) and our shaded netty is Netty 4 (io.netty).